### PR TITLE
Fix the Type overflow. #1

### PR DESCRIPTION
**Pull request by anuragdogra2192**

### CODE REVIEW

File: src/definition_reader.cpp [hunk 1]
```diff
@@ -477,8 +477,9 @@ void DefinitionReader::handleDefinition(SystemTreeNode &sysTreeNode) {
 
   else {
     if (sysTreeNode.s_parent == OTF2_UNDEFINED_SYSTEM_TREE_NODE) {
-      auto parent_id(m_traceDefs.verifyAbstractParent(sysTreeNode.s_name));
-      sysTreeNode.s_parent = parent_id;
+      auto status(m_traceDefs.verifyAbstractParent(sysTreeNode.s_name));
+      if (status == true)
```
Comment (harryherold) [inline src/definition_reader.cpp]: Use braces also for one-liners.

File: src/definition_store.cpp [hunk 2]
```diff
@@ -142,14 +142,14 @@ bool DefinitionStore::hasAbstractNode() {
   return false;
 }
 
-size_t DefinitionStore::verifyAbstractParent(const size_t sysNodeName) {
+bool DefinitionStore::verifyAbstractParent(const size_t sysNodeName) {
```
Comment (harryherold) [inline src/definition_store.cpp]: `verifyAbstractParent` returns true what does it mean? So your method name should also fit to the return!
e.g. `isEmpty`

File: src/maps.cpp [hunk 3]
```diff
@@ -128,8 +128,7 @@ void Maps::insertIoFileProperty(
 }
 
 void Maps::insertIoFileDefinitionToDatabase() {
-  vector<IoFileDefinition>::iterator iter;
-  for (iter = m_ioFileDefinition.begin(); iter != m_ioFileDefinition.end();
+  for (auto iter = m_ioFileDefinition.begin(); iter != m_ioFileDefinition.end();
```
Comment (harryherold) [inline src/maps.cpp]: The purpose of the pull request is different.

File: src/definition_reader.cpp (hunk 1 above)
Comment (harryherold) [inline src/definition_reader.cpp]: useless. just `if(status)`

File: src/definition_reader.cpp [hunk 4]
```diff
@@ -477,8 +477,9 @@ void DefinitionReader::handleDefinition(SystemTreeNode &sysTreeNode) {
 
   else {
     if (sysTreeNode.s_parent == OTF2_UNDEFINED_SYSTEM_TREE_NODE) {
-      auto parent_id(m_traceDefs.verifyAbstractParent(sysTreeNode.s_name));
-      sysTreeNode.s_parent = parent_id;
+      auto status(m_traceDefs.verifyAbstractParent(sysTreeNode.s_name));
```
Comment (harryherold) [inline src/definition_reader.cpp]: Why the constructor for bool?

**Review by harryherold — CHANGES_REQUESTED** [review]

File: src/definition_reader.cpp [hunk 5]
```diff
@@ -477,9 +477,10 @@ void DefinitionReader::handleDefinition(SystemTreeNode &sysTreeNode) {
 
   else {
     if (sysTreeNode.s_parent == OTF2_UNDEFINED_SYSTEM_TREE_NODE) {
-      auto status(m_traceDefs.verifyAbstractParent(sysTreeNode.s_name));
-      if (status == true)
+      bool status(m_traceDefs.requiresAbstractParent(sysTreeNode.s_name));
```
Comment (harryherold) [inline src/definition_reader.cpp]: From first review > Why the constructor for bool?

**Review by harryherold — CHANGES_REQUESTED** [review]

File: src/definition_reader.cpp (hunk 5 above)
Comment (anuragdogra2192) [reply src/definition_reader.cpp]: I will fix it asap.

Comment (harryherold) [reply src/definition_reader.cpp]: and the answer?

Comment (harryherold) [reply src/definition_reader.cpp]: Why not:
```
sysTreeNode.s_parent = m_traceDefs.getParent(sysTreeNode.s_name);
```

Comment (anuragdogra2192) [reply src/definition_reader.cpp]: In definition_reader.cpp
```proper implementation
 if (sysTreeNode.s_parent == OTF2_UNDEFINED_SYSTEM_TREE_NODE) {
      sysTreeNode.s_parent = m_traceDefs.getAbstractParent(sysTreeNode.s_name);
    }
```
Respectively in definition_store.cpp
```proper implementation
uint32_t DefinitionStore::getAbstractParent(const size_t sysNodeName) {
  if (m_sysTreeNodeDb.size() > 1) {
    if (m_sysTreeNodeDb[1].s_name != sysNodeName) {
      m_sysTreeNodeDb[0].s_newRootparent = true;
      return 0;
    }
  }
  return OTF2_UNDEFINED_SYSTEM_TREE_NODE;
}
```

**Review by harryherold — APPROVED** [review]

**Pull request merged**